Dockerfile

```diff
@@ -17,7 +17,7 @@ ARG AZURITE_VERSION=3.18.0
 ARG GOTESTSUM_VERSION=v1.9.0
 ARG DELVE_VERSION=v1.21.0
 
-ARG GO_VERSION=1.20
+ARG GO_VERSION=1.21
```

Comment:
`go.mod` should also be updated.

Reply:
If it still works on go1.20, Id keep it on the lowest version supported; go started to enforce versions based on go.mod (or download tool chains)

Reply:
^ for go.mod that is

Reply:
I think if we can it would be good to keep them in sync - otherwise, all this PR does is just keep using 1.20, by downloading that toolchain?

Reply:
we should probably set GOTOOLCHAIN=local to have a predictable version, but don't think it downgrades versions

Reply:
I don't think we need to, already set in base image: https://github.com/docker-library/golang/blob/f0d301c385958888968d6ab8f8510c3cd5cce704/1.21/bookworm/Dockerfile#L127-L129